src/test/scala/io/iohk/ethereum/blockchain/sync/PeersClientSpec.scala

```diff
@@ -0,0 +1,61 @@
+package io.iohk.ethereum.blockchain.sync
+
+import java.net.InetSocketAddress
+
+import akka.actor.ActorSystem
+import akka.testkit.TestProbe
+import akka.util.ByteString
+import io.iohk.ethereum.network.EtcPeerManagerActor.PeerInfo
+import io.iohk.ethereum.network.Peer
+import io.iohk.ethereum.network.p2p.messages.CommonMessages.Status64
+import io.iohk.ethereum.network.p2p.messages.Versions
+import org.scalatest.flatspec.AnyFlatSpec
+import org.scalatest.matchers.should.Matchers
+import org.scalatestplus.scalacheck.ScalaCheckPropertyChecks
+
+class PeersClientSpec extends AnyFlatSpec with Matchers with ScalaCheckPropertyChecks {
+  "PeerClient" should "determined the best peer based on it latest checkpoint number and total difficulty" in {
+    import Peers._
+
+    val table = Table[Map[Peer, PeerInfo], Option[Peer]](
+      ("PeerInfo map", "Expected best peer"),
```

Comment:
Minor, should we add some comment here on why each peer is selected in every situation?